src/background/messaging/HttpConnection.ts

```diff
@@ -9,31 +9,22 @@ import {
   formatJsonRpcError,
   isJsonRpcRequest,
 } from '@walletconnect/jsonrpc-utils';
-import { networksStore } from 'src/modules/networks/networks-store.background';
-import type { ChainId } from 'src/modules/ethereum/transactions/ChainId';
 
 export class HttpConnection extends EventEmitter {
```

Comment:
Looks like HttpConnection is not really living up to its potential—it's short-lived and no one subscribes to its `'payload'` events. Given that, do we really need it to be a class? If we're not leveraging inheritance features or maintaining state, we might be better off with a more straightforward approach. How about refactoring it into a simple async function like `sendJsonRpcRequest(url: string, request: JsonRpcPayload)`?

Reply:
That's true. It's an artifact of an earlier implementation. I wouldn't refactor it in this PR though